modules/layers/src/icon-layer/icon-layer.js

```diff
@@ -180,7 +183,10 @@ export default class IconLayer extends Layer {
         Object.assign({}, uniforms, {
           iconsTexture,
           iconsTextureDim: [iconsTexture.width, iconsTexture.height],
-          sizeScale
+          sizeScale,
+          sizeMinPixels,
+          sizeMaxPixels,
+          sizeInMeters: sizeUnits === 'meters'
```

Comment:
Just amend `sizeScale` here and avoid doing this in the shader for every vertex:

```
sizeScale: sizeScale * (sizeUnits === 'pixels' ? this.context.viewport.distanceScales.metersPerPixel[2] : 1)
```